### DISCUSSION TITLE
HDMI/KVM/Monitor input switch support

### DISCUSSION THREAD

**Discussion by kohend** [input-leap/input-leap#1926] [Development of Input Leap]
Hi,
I want to implement a better version of the #758 PR, I already wrote the code that runs a command on screen switch and it works well with a small script using ddcutil, but I need some help figuring out how to do some additional things:
- How do I link the checkbox to the input field in the settings dialog?
- It seems when the server is locked, there's a switch that is not from the switchScreen method in the Sever file, where is it?

**Comment by shymega** [comment]
This is out of the scope of the project. It's always been a software KVM, not for monitor switching. We do not have a hook system.

I would rather this functionality remains out of scope.

**Comment by kohend** [comment]
The V in the acronym KVM is video, AFAIK, the point is to use multiple computers with the same set of keyboard, mouse and monitor, right now, you need multiple monitors (which physically I have no room for), this allows me to use a single monitor, just like a physical KVM does.
It could be used for USB switches as well, why shouldn't it be supported in Input-leap?
And the only change in the software is adding a config for a command and running it, it doesn't do the switching by itself.

**Reply by shymega** [reply]
Video, yes, but we don't do that. That's a misnomer when it comes to Input Leap.

We could look into integrating hooks into the Event handler subsystem in the future, but that would require a tracking issue.

We have had requests (and a PR) for hooks before.

**Reply by shymega** [reply]
See #758.

**Reply by kohend** [reply]
I know you had, I mentioned it in the first post in the discussion.
I wanted some help in how to make it, or should I push the WIP hook and get comments?

**Reply by shymega** [reply]
You can make a PR, but you won't be able to push to #758. If you add 'Closes #758' to your PR, that would be of usefulness.
